### PR TITLE
[ML] Check range in bounds when clearing recycled models

### PR DESCRIPTION
**Pull request by tveasey**
Add failsafes to ensure that we never try and reset state for identifiers outside vector bounds. This shouldn't happen, but this is the likely cause of #76 and there are no side effects from checking and issuing an error in such cases. Investigating and fixing the root cause of #76 will be the subject of separate PR.

### CODE REVIEW

File: lib/model/CPopulationModel.cc [hunk 1]
```diff
@@ -445,16 +445,24 @@ void CPopulationModel::createNewModels(std::size_t n, std::size_t m) {
...
-            m_PersonAttributeBucketCounts[cid] = *m_NewPersonBucketCounts;
+        if (cid < m_AttributeFirstBucketTimes.size()) {
+            m_AttributeFirstBucketTimes[cid] = CAnomalyDetectorModel::TIME_UNSET;
+            m_AttributeLastBucketTimes[cid] = CAnomalyDetectorModel::TIME_UNSET;
+            m_DistinctPersonCounts[cid] = m_NewDistinctPersonCounts;
+            if (m_NewPersonBucketCounts) {
+                m_PersonAttributeBucketCounts[cid] = *m_NewPersonBucketCounts;
+            }
+        } else {
+            LOG_ERROR("Recycled attribute identifier '"
```
Comment (droberts195) [inline lib/model/CPopulationModel.cc]: There's a missing `<<` after the `LOG_ERROR(`.

Also, have you run `clang-format` on this file?  CI is set to fail now if the formatting isn't what `clang-format` would choose.

**Comment by droberts195** [comment]
I think it would be good to make the same fix in 5.6.10 as well, since that was where the problem was reported, and some people will be using 5.6 until early next year.

File: lib/model/CAnomalyDetectorModel.cc [hunk 2]
```diff
@@ -448,7 +448,12 @@ void CAnomalyDetectorModel::createNewModels(std::size_t n, std::size_t /*m*/) {
 void CAnomalyDetectorModel::updateRecycledModels() {
     TSizeVec& people{m_DataGatherer->recycledPersonIds()};
     for (auto pid : people) {
-        m_PersonBucketCounts[pid] = 0.0;
+        if (pid < m_PersonBucketCounts.size()) {
+            m_PersonBucketCounts[pid] = 0.0;
+        } else {
+            LOG_ERROR("Recycled person identifier '"
```
Comment (edsavage) [inline lib/model/CAnomalyDetectorModel.cc]: Missing '<<' at the start of the error message

**Comment by tveasey** [comment]
Agreed Dave, I was planning to make the necessary changes in that repo too.

**Pull request merged**